provider/disk_utils.py

```diff
@@ -0,0 +1,132 @@
+    disk in Windows ones.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param did: The ID of the VM's disk.
+    :param image_name: The image name. e.g: stg0
+    :param dst_dir: The destination directory for mounting.
+    :param fstype: The kind of filesystem (ntfs or xfs).
+    :returns: mount point for windows or folder name in Linux
+    """
+    session = vm.wait_for_login()
+    is_windows = params.get("os_type") == "windows"
+    image_params = params.object_params(image_name)
+    img_size = image_params.get("image_size")
+
+    if fstype is None:
+        fstype = "ntfs" if is_windows else "xfs"
+
+    if not dst_dir and not is_windows:
+        dst_dir = f"/mnt/{image_name}"
+        session.cmd(f"mkdir -p {dst_dir}")
+
+    if is_windows:
+        utils_disk.update_windows_disk_attributes(session, did)
+        utils_disk.clean_partition_windows(session, did)
+        return utils_disk.configure_empty_disk(session, did, img_size, "windows")[0]
+    else:
+        # Send only the disk ID
+        disk_id = did.split("/")[-1]
+        utils_disk.create_filesyetem_linux(session, disk_id, fstype)
+        if utils_disk.mount(did, dst_dir, fstype, session=session):
+            return dst_dir
+
+
+def execute_io_test(
+    params,
+    vm,
+    image,
+    serial,
+    fstype=None,
+    dst_dir=None,
+    io_command=None,
+    clean=True,
+    ignore_all_errors=False,
+    timeout=60,
+):
+    """
+    Execute the I/O write test on the disk image inside the guest.
+    Note: The target disk image should be the uninitialized disk.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param image: The image name. e.g: stg0
+    :param serial: The serial number of the disk
+    :param dst_dir: The destination directory for I/O write testing.
+    :param io_command: The command for I/O test.
+    :param clean: If True, cleans the environment
+    :param ignore_all_errors: Checks the errors of the I/O command
+    :param timeout: The timeout for the I/O command.
+    """
+    session = vm.wait_for_login()
+    os_type = params.get("os_type")
+    is_windows = True if os_type == "windows" else False
+
+    did = get_disk_by_serial(os_type, serial, session)
+    mount_point = init_disk_by_id(params, vm, did, image, dst_dir, fstype)
+
+    log_message = f"The did: {did} and the mount point: {mount_point}"
+    LOG_JOB.debug(log_message)
+
+    try:
+        # Start to run I/O command
+        if io_command:
+            io_cmd = io_command % mount_point
+            session.cmd(io_cmd, timeout, ignore_all_errors=ignore_all_errors)
+    finally:
+        if clean:
+            # We need to try to clean up and roll back the environment finally.
+            if not is_windows:
+                utils_disk.is_mount(did, mount_point, fstype=fstype, session=session)
+                utils_disk.umount(did, mount_point, fstype=fstype, session=session)
```

Comment:
```suggestion
                if utils_disk.is_mount(did, mount_point, fstype=fstype, session=session)
                    utils_disk.umount(did, mount_point, fstype=fstype, session=session)
```
I think that the `unmount` should be executed with the condition of `is_mount is Ture`.

Reply:
Done !